.ci/azure-pipelines/docs-pipeline.yaml

```diff
@@ -0,0 +1,50 @@
+trigger:
+  paths:
+    include:
+    - doc
+    - README.md
+    - CHANGES.md
+    - CONTRIBUTING.md
+
+pr:
+  paths:
+    include:
+    - doc
+    - README.md
+    - CHANGES.md
+    - CONTRIBUTING.md
+
+resources:
+  pipelines:
+    - pipeline: Build-CI
+      source: Build
+      trigger:
+        stages:
+        - build_gcc
+  containers:
+    - container: fmt # for formatting.yaml
+      image: pointcloudlibrary/fmt
+    - container: doc # for documentation.yaml
+      image: pointcloudlibrary/doc
+    - container: env1804 # for tutorials.yaml
+      image: pointcloudlibrary/env:18.04
+
+stages:
+  - stage: formatting
+    displayName: Formatting
+    # if docs pipeline triggered by build_gcc stage,
```

Comment:
The build gcc step hasn't changed. Why will it trigger the docs pipeline?

Reply:
Yes, it won't trigger `docs-pipeline` in docs changes. And I add the `condition` specially for code changes.

`docs-pipeline` triggers:
- by `build_gcc` stage only in code changes, so `formatting` has already run in `main-pipeline`, and it won't run here twice.
- by Push or PR only in docs changes(I include docs in `doc-pipeline`, and exclude them in `main-pipeline`), so `formatting` is ready to run in `docs-pipeline`.